domain/src/Aggregates/SharePoolingAsset/SharePoolingAsset.php

```diff
@@ -148,37 +140,29 @@ private function replayDisposals(SharePoolingAssetDisposals $disposals): void
         }
     }
 
-    private function recordDisposal(DisposeOfSharePoolingAsset $action): void
-    {
-        $sharePoolingAssetDisposal = DisposalProcessor::process(
-            disposal: $action,
-            transactions: $this->transactions,
-        );
-
-        $this->recordThat(new SharePoolingAssetDisposedOf(disposal: $sharePoolingAssetDisposal));
-    }
-
     public function applySharePoolingAssetDisposedOf(SharePoolingAssetDisposedOf $event): void
     {
+        // Adjust quantities for acquisitions whose quantities were allocated to the disposal
+        QuantityAdjuster::applyDisposal($event->disposal, $this->transactions);
+
         $this->previousTransactionDate = $event->disposal->date;
         $this->transactions->add($event->disposal);
     }
 
     private function revertDisposals(SharePoolingAssetDisposals $disposals): void
     {
         foreach ($disposals as $disposal) {
-            // Restore quantities deducted from the acquisitions whose quantities were allocated to the disposal
-            QuantityAdjuster::revertDisposal($disposal, $this->transactions);
-
             $this->recordThat(new SharePoolingAssetDisposalReverted(disposal: $disposal));
         }
     }
 
     public function applySharePoolingAssetDisposalReverted(SharePoolingAssetDisposalReverted $event): void
     {
-        // Replace the disposal in the array with the same disposal, but with reset quantities. This
-        // way, when several disposals are being replayed, a disposal won't be matched with future
-        // acquisitions within the next 30 days if these acquisitions have disposals on the same day
+        // Restore quantities deducted from acquisitions whose quantities were allocated to the disposal
+        QuantityAdjuster::revertDisposal($event->disposal, $this->transactions);
```

Comment:
ℹ️ This is now done upon applying the reversion event, meaning the quantities are always right